bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -399,26 +400,26 @@ void computeRuns (GC gc) {
 			if (length > 1) {
 				int stop = j + length - 1;
 				if (stop < tabsLength) {
-					run.width += tabs[stop] - tabs[j];
+					run.width += DPIUtil.autoScaleUp(getDevice(), tabs[stop], gc.data.shell) - DPIUtil.autoScaleUp(getDevice(), tabs[j], gc.data.shell);
```

Comment:
here you can definitely do 
DPIUtil.autoScaleUp(getDevice(), tabs[stop] -tabs[j], gc.data.shell)